src/foam/u2/detail/SectionedDetailPropertyView.js

```diff
@@ -237,8 +237,7 @@ foam.CLASS({
 
           var isCheckBox = false;
           if ( foam.Function.isInstance(prop.view) ) {
-            var view = prop.view(null, this.__subSubContext__);
-            isCheckBox = view.class === 'foam.u2.CheckBox' || ( view.cls_ && view.cls_.id === 'foam.u2.CheckBox' );
+            isCheckBox = prop.view.class === 'foam.u2.CheckBox' || ( prop.view.cls_ && prop.view.cls_.id === 'foam.u2.CheckBox' );
```

Comment:
Doesn't work for subclasses. Better would be to use:
foam.u2.CheckBox.isSubClass().
Although, this whole design isn't ideal in the first place, since hard-coding checkbox support into a detailView isn't extensible.